front/src/components/BeeHive/BeehiveForm.tsx

```diff
@@ -0,0 +1,114 @@
+import { FormEvent, useState } from 'react';
+import { FormErrorsMap } from '@app/components/UI/Form';
+import usePreviousUrlFromLocation from '@app/hooks/usePreviousUrlLocationState.tsx';
+import { Button, Stack } from '@mantine/core';
+import { trans } from '@app/translations';
+import { FormRootErrors } from '@app/pages/Error/FormRootErrors.tsx';
+import CompactTextInput from '@app/components/UI/CompactInput/CompactTextInput.tsx';
+import { Beehive } from '@app/models/types/Beehive';
+import { useParams } from 'react-router-dom';
+import TopNavigationMenu from '../UI/TopNavigation/TopNavigationMenu';
+import ApiaryHome from '@app/pages/Apiary/ApiaryHome';
+import ApiaryList from '@app/pages/Apiary/ApiaryList';
+import { route } from '@app/router/generator';
+import CompactSelect from '../UI/CompactInput/CompactSelect';
+import BeehiveHome from '@app/pages/Beehive/BeehiveHome';
+
+
+
+export interface BeehiveData {
+  name: string | null
+  bee: string | null
+  age: number | null
+  apiary: string | null
+}
+
+interface Props {
+  onSubmit: (payload: BeehiveData) => void
+  errors?: FormErrorsMap<keyof BeehiveData>
+}
+
+interface CreateProps extends Props {
+  initialData?: never
+}
+
+interface UpdateProps extends Props {
+  initialData: Beehive
+}
+
+export default function BeehiveForm({
+  onSubmit,
+  initialData = undefined,
+  errors = {},
+}: CreateProps | UpdateProps) {
+
+  const { uid } = useParams();
+  const isUpdate = initialData !== undefined;
+  const { previousUrl } = usePreviousUrlFromLocation();
+  const beeTypes = ['Black', 'Italian', 'Caucasian', 'Carnolien', 'Buckfast'];
```

Comment:
Should pull this from the backend, not hardcode it.
I believe the function is called `listBeeType` in the backend, you should look into this.